.github/actions/test-report/action.yml

```diff
@@ -59,7 +60,19 @@ inputs:
 runs:
   using: "composite"
   steps:
+    - name: Report go test results
+      if: contains('go-json', inputs.reporter)
+      uses: elastic/apm-pipeline-library/.github/actions/test-go-report@feature/go-json-reporter
```

Comment:
IIRC to pass an expression to `uses`, if so when we have more than two alternatives we probably want to change to have a step that selects the action and puts the name in an output we use here.